scripts/globals/weaponskills.lua

```diff
@@ -216,51 +215,35 @@ local function getMultiAttacks(attacker, target, wsParams)
-        elseif math.random(1, 100) <= doubleRate then
-            bonusHits = bonusHits + 1
-        elseif i == 1 and math.random(1, 100) <= oaThriceRate then -- Can only proc on first hit
-            bonusHits = bonusHits + 2
-        elseif i == 1 and math.random(1, 100) <= oaTwiceRate then -- Can only proc on first hit
-            bonusHits = bonusHits + 1
-        end
-
-        if i == 1 then
-            attacker:delStatusEffect(xi.effect.ASSASSINS_CHARGE)
-            attacker:delStatusEffect(xi.effect.WARRIORS_CHARGE)
-
-            -- recalculate DA/TA/QA rate
-            doubleRate = attacker:getMod(xi.mod.DOUBLE_ATTACK) + attacker:getMerit(xi.merit.DOUBLE_ATTACK_RATE)
-            tripleRate = attacker:getMod(xi.mod.TRIPLE_ATTACK) + attacker:getMerit(xi.merit.TRIPLE_ATTACK_RATE)
-            quadRate   = attacker:getMod(xi.mod.QUAD_ATTACK)
+    if math.random(1, 100) <= quadRate then
+        bonusHits = bonusHits + 3
+    elseif math.random(1, 100) <= tripleRate then
+        bonusHits = bonusHits + 2
+    elseif math.random(1, 100) <= doubleRate then
+        bonusHits = bonusHits + 1
+    elseif firstHit and math.random(1, 100) <= oaThriceRate then -- Can only proc on first hit
+        bonusHits = bonusHits + 2
+    elseif firstHit and math.random(1, 100) <= oaTwiceRate then -- Can only proc on first hit
+        bonusHits = bonusHits + 1
+    end
+
+    attacker:delStatusEffect(xi.effect.ASSASSINS_CHARGE)
+    attacker:delStatusEffect(xi.effect.WARRIORS_CHARGE)
+
+    -- Try OaX for Jumps
+    -- ... What's the correct dual wield interaction?
+    if isJump and bonusHits == 0 and attacker:isPC() then
+        -- getWeaponHitCount will always return 1 if there's a weapon in the slot, which is already accounted for.
+        if offHand then
+            bonusHits = attacker:getWeaponHitCount(true) - 1
+        else
+            bonusHits = attacker:getWeaponHitCount(false) - 1
         end
     end
 
-    -- Only get an additional offHand hit with H2H if the wsParams.numHits == 1
-    -- "Note that Hand-to-Hand weaponskills' listed hits do include the offhand hit. However, if a weaponskill does not list multiple hits, it still has an additional offhand hit."
-    -- see "hits" column: https://wiki-ffo-jp.translate.goog/html/19049.html?_x_tr_sch=http&_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=it&_x_tr_pto=wapp
-    -- basically h2h always gets 2 "initial" hits, but `numHits` is hard respected for the extra hits
-    -- whereas dual wiedling gets 2 "initial" hits, but `numHits` for e.g. dancing edge is respected as though only one "initial" was performed
-    -- tl;dr: DW gives an extra WS hit everytime, H2H simply ensures every WS gets at least 2 hits (yes backhand blow and dragon kick are both 2-hit ws...)
     numHits = numHits + bonusHits
-    if attacker:getOffhandDmg() > 0 then
-        numHits = numHits + 1
-    end
 
-    -- cap total hits at 8
-    return utils.clamp(numHits, 1, 8)
+    return numHits
```

Comment:
No clamp needed?

Reply:
Clamp is moved to the hit count loops, it's not possible for that function to return > 4 anyway